mods/bones/init.lua

```diff
@@ -190,15 +204,68 @@ local function is_all_empty(player_inv)
 	return true
 end
 
+--functions registered this way won't becalled if bones_mode is keep
+function bones.public.register_transfer_inventory_to_bones_on_player_death(func)
+	bones.private.dead_player_callbacks[#(bones.private.dead_player_callbacks)]=func
```

Comment:
Use `table.insert` instead. This is completely broken in its current form: It just sets `dead_player_callbacks[0] = func`.